test/rotation.jl

```diff
@@ -0,0 +1,79 @@
+function rotation_testsuite(Backend)
+    device(x) = adapt(Backend(), x)
+    gradtest_fn = Backend == CPU ? gradtest : gputest
+    T = Float32
+    atol = T == Float32 ? 1e-3 : 1e-6
+    rtol = T == Float32 ? 1f-2 : 1f-6
+    angles = deg2rad.([0, 0.0001, 35, 90, -90, -90.0123, 170, 180, 270, 360, 450, 1234.1234]) 
+
+    @testset "imrotate" begin
+        @testset "Simple test" begin
+            arr = device(zeros((6, 6, 1, 1))); 
+            arr[3:4, 4, 1, 1] .= 1;
+            @test all(cpu(NNlib.imrotate(arr, deg2rad(45))) .≈ [0.0 0.0 0.0 0.0 0.0 0.0; 0.0 0.0 0.0 0.0 0.0 0.0; 0.0 0.0 0.0 0.29289321881345254 0.585786437626905 0.0; 0.0 0.0 0.08578643762690495 1.0 0.2928932188134524 0.0; 0.0 0.0 0.0 0.08578643762690495 0.0 0.0; 0.0 0.0 0.0 0.0 0.0 0.0])
+        end
+
+
+        @testset "Compare with ImageTransformations" begin
+            for sz in [(51,51,1,1), (52,52,1,1)]
```

Comment:
Oh and what about images of odd and even size like (51,52)? Testing thoroughly is really painful :D

Reply:
There's indeed some tests failing in the case of nearest neighbour interpolation.

It's just at some angles and seems more like some rounding issues to me. So not entirely sure how to treat it.